### PR TITLE
feat: add typescript support

### PR DESCRIPTION
**Pull request by turneand**
As per #20 , adding typescript support.

### CODE REVIEW

File: fastify-opentelemetry.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+/**
+ * Options for the OpenTelemetry plugin.
+ */
+export interface OpenTelemetryPluginOptions {
+  serviceName?: string,
+  exposeApi?: boolean,
+  formatSpanName?: (serviceName: string, raw: FastifyRequest) => string,
```
Comment (HW13) [inline fastify-opentelemetry.d.ts:20]: The `raw` param isn't the entire `FastifyRequest`, it's just the `request.raw` property. Using node's `Http2ServerRequest` interface should work.

**Review by HW13 — CHANGES_REQUESTED** [review]
Thanks for the contribution @turneand! A few tweaks, and this should be good to go 👍

File: fastify-opentelemetry.d.ts (hunk 1 above)
Comment (turneand) [reply fastify-opentelemetry.d.ts:20]: Will switch out to use:
```
formatSpanName?: (serviceName: string, raw: FastifyRequest['raw']) => string,
```

File: fastify-opentelemetry.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  formatSpanAttributes?: {
+    request?: (request: FastifyRequest) => object,
+    reply?: (reply: FastifyReply) => object,
+    error?: (error: Error) => object,
```
[the remark this answers is not in the document]
Comment (turneand) [reply fastify-opentelemetry.d.ts:24]: Ah, apologies, yes, absolutely right.  I presume that was just a placeholder whilst I put the rest together

File: fastify-opentelemetry.d.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { FastifyPluginCallback, FastifyReply, FastifyRequest } from 'fastify';
+import { context, getSpan, propagation, TextMapGetter, TextMapSetter, Tracer } from "@opentelemetry/api"
+
+declare module 'fastify' {
+  interface FastifyRequest {
+    activeSpan: () => ReturnType<typeof getSpan>,
+    context: () => ReturnType<typeof context.active>,
+    tracer: () => Tracer,
+    inject: <Carrier> (carrier: Carrier, setter?: TextMapSetter) => ReturnType<typeof propagation.inject>,
+    extract: <Carrier> (carrier: Carrier, getter?: TextMapGetter) => ReturnType<typeof propagation.extract>,
```
[the remark this answers is not in the document]
Comment (turneand) [reply fastify-opentelemetry.d.ts:10]: Hmm, no idea how I managed to mess that one up quite so much!
I'll remove all of the "ReturnType" magic and simplify accordingly.

One thing to point out now though is that Span can be undefined, so will do the rest and set:
```
activeSpan: Span | undefined,
```

I've also realized I should mark all of them as readonly!

File: test/types/fastify-opentelemetry.test-d.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import fastify, { FastifyRequest } from "fastify";
+import { expectType } from 'tsd'
+import { Context, Span, Tracer } from "@opentelemetry/api";
+
+// should be able to use openTelemetry functions on FastifyRequest
+fastify().get('/', (req, res) => {
+  expectType<OpenTelemetryReqInstance>(req.openTelemetry());
+  expectType<Span | undefined>(req.openTelemetry().activeSpan);
+  expectType<Context>(req.openTelemetry().context);
+  expectType<Tracer>(req.openTelemetry().tracer);
```
Comment (HW13) [inline test/types/fastify-opentelemetry.test-d.ts:11]: Could we also add expects for `inject` and `extract` return values?
```ts
expectType<void>(req.openTelemetry().inject({}))
expectType<Context>(req.openTelemetry().extract({}))
```

**Review by HW13 — COMMENTED** [review]
Just one last nit, and I think we're good 👍

File: test/types/fastify-opentelemetry.test-d.ts (hunk 4 above)
Comment (turneand) [reply test/types/fastify-opentelemetry.test-d.ts:11]: Added...   thank you for taking the time to review these, it's much appreciated!

**Pull request merged**